src/main/java/io/github/wimdeblauwe/hsbt/mvc/HtmxResponse.java

```diff
@@ -0,0 +1,94 @@
+package io.github.wimdeblauwe.hsbt.mvc;
+
+import org.springframework.util.Assert;
+
+import java.util.*;
+
+/**
+ * Representation of HTMX partials.
+ *
+ * @author Oliver Drotbohm
+ * @author Clint Checketts
+ */
+final public class HtmxResponse {
+
+	private final Collection<String> templates;
+	private final Map<String, String> triggers;
+	private final Map<String, String> triggersAfterSettle;
+	private final Map<String, String> triggersAfterSwap;
+
+	public HtmxResponse() {
+        this.templates = new ArrayList<>();
+        this.triggers = new HashMap<>();
+        this.triggersAfterSettle = new HashMap<>();
+        this.triggersAfterSwap = new HashMap<>();
+	}
+	
+	/**
+	 * Append the rendered fragment.
+	 *
+	 * @param template must not be {@literal null} or empty.
+	 * @return same HtmxResponse for chaining
+	 */
+	public HtmxResponse addTemplate(String template) {
```

Comment:
We could create a value object `TemplateReference` that has a few factory methods like:

* `TemplateReference.of("mytemplate")`
* `TemplateReference.of("mytemplate :: myfragment")`
* `TemplateReference.of("mytemplate", "myfragment")`

It could throw an exception when having a `::` in the 2 parameter version.

Then add an overload to `addTemplate()` to use such a `TemplateReference` ?

Reply:
@odrotbohm As this PR is now merged, you can create a new issue if you would still like to have this.